core/genesis.go

```diff
@@ -282,8 +282,9 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *triedb.Database, g
 			if overrides != nil && overrides.OverrideOptimismCanyon != nil {
 				config.CanyonTime = overrides.OverrideOptimismCanyon
 				config.ShanghaiTime = overrides.OverrideOptimismCanyon
-				if config.Optimism != nil && config.Optimism.EIP1559DenominatorCanyon == 0 {
-					config.Optimism.EIP1559DenominatorCanyon = 250
+				if config.Optimism != nil && *config.Optimism.EIP1559DenominatorCanyon == 0 {
+					eip1559DenominatorCanyon := uint64(250)
+					config.Optimism.EIP1559DenominatorCanyon = &eip1559DenominatorCanyon
```

Comment:
Shouldn't we check that `*config.Optimism.EIP1559DenominatorCanyon != nil` as well, or risk a panic here?

Reply:
Changed the code so that if the `overrides.OverrideOptimismCanyon` field is set, this value will always get set to 250

Reply:
Changed logic here https://github.com/ethereum-optimism/op-geth/pull/345/commits/a850e59f45a9f1ad81dd7faf72214dc89f840088